# Data processing for US Airport Dataset

Our data for analysis comes from [Kaggle US Airport Dataset](https://www.kaggle.com/datasets/flashgordon/usa-airport-dataset). 

As our main focus is on implementing and comparing algorithms that extract useful information from the dataset, we are going to pre-process the data in a way that
columns that contain unused information and duplicate rows are removed.

In [1]:
import pandas as pd

In [26]:
raw_data = pd.read_csv('raw_usa-airport-dataset.csv')

## Overview of the raw dataset

In [27]:
raw_data.shape

(3606803, 15)

In [28]:
raw_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3606803 entries, 0 to 3606802
Data columns (total 15 columns):
 #   Column                  Dtype  
---  ------                  -----  
 0   Origin_airport          object 
 1   Destination_airport     object 
 2   Origin_city             object 
 3   Destination_city        object 
 4   Passengers              int64  
 5   Seats                   int64  
 6   Flights                 int64  
 7   Distance                int64  
 8   Fly_date                object 
 9   Origin_population       int64  
 10  Destination_population  int64  
 11  Org_airport_lat         float64
 12  Org_airport_long        float64
 13  Dest_airport_lat        float64
 14  Dest_airport_long       float64
dtypes: float64(4), int64(6), object(5)
memory usage: 412.8+ MB


In [29]:
raw_data.isnull().sum()

Origin_airport               0
Destination_airport          0
Origin_city                  0
Destination_city             0
Passengers                   0
Seats                        0
Flights                      0
Distance                     0
Fly_date                     0
Origin_population            0
Destination_population       0
Org_airport_lat           6954
Org_airport_long          6954
Dest_airport_lat          6807
Dest_airport_long         6807
dtype: int64

## Clean up dataset

The resultant dataset size shrinks from 3606803 to 34775.

In [62]:
# drop NA values
data = raw_data.dropna()

In [65]:
# remove duplicate entries
data['Combined'] = data[['Origin_airport', 'Destination_airport']].apply(
    lambda x: ', '.join(x), axis=1)

/var/folders/gn/sdx6p9hx42v35r6bwj_s2pfw0000gn/T/ipykernel_95964/3363548380.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data['Combined'] = data[['Origin_airport', 'Destination_airport']].apply(


In [70]:
data.drop_duplicates(subset=['Combined'], inplace=True)

/var/folders/gn/sdx6p9hx42v35r6bwj_s2pfw0000gn/T/ipykernel_95964/2870002013.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data.drop_duplicates(subset=['Combined'], inplace=True)


In [74]:
# drop unused columns
data.drop(['Origin_city', 'Destination_city',
           'Passengers', 'Seats', 'Flights', 'Fly_date', 'Origin_population',
           'Destination_population', 'Combined'], axis=1, inplace=True)


/var/folders/gn/sdx6p9hx42v35r6bwj_s2pfw0000gn/T/ipykernel_95964/3323893814.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data.drop(['Origin_city', 'Destination_city',


In [75]:
data.shape

(34775, 7)

In [76]:
data.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 34775 entries, 1 to 3606801
Data columns (total 7 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   Origin_airport       34775 non-null  object 
 1   Destination_airport  34775 non-null  object 
 2   Distance             34775 non-null  int64  
 3   Org_airport_lat      34775 non-null  float64
 4   Org_airport_long     34775 non-null  float64
 5   Dest_airport_lat     34775 non-null  float64
 6   Dest_airport_long    34775 non-null  float64
dtypes: float64(4), int64(1), object(2)
memory usage: 2.1+ MB


In [77]:
data.head(5)

Origin_airport Destination_airport  Distance  Org_airport_lat  \
1             EUG                 RDM       103        44.124599   
4             MFR                 RDM       156        42.374199   
9             SEA                 RDM       228        47.449001   
14            PDX                 RDM       116        45.588699   
38            LMT                 RDM       147        42.156101   

    Org_airport_long  Dest_airport_lat  Dest_airport_long  
1        -123.211998         44.254101        -121.150002  
4        -122.873001         44.254101        -121.150002  
9        -122.308998         44.254101        -121.150002  
14       -122.598000         44.254101        -121.150002  
38       -121.733002         44.254101        -121.150002

## Dataset 1: from which node-edge graph can be built

Airports are nodes and the distances between airports are edges.

In [81]:
graph_data = data.drop(
    ['Org_airport_lat', 'Org_airport_long', 'Dest_airport_lat', 'Dest_airport_long'],
    axis=1)

Origin_airport Destination_airport  Distance
1             EUG                 RDM       103
4             MFR                 RDM       156
9             SEA                 RDM       228
14            PDX                 RDM       116
38            LMT                 RDM       147

In [105]:
graph_data.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 34775 entries, 1 to 3606801
Data columns (total 3 columns):
 #   Column               Non-Null Count  Dtype 
---  ------               --------------  ----- 
 0   Origin_airport       34775 non-null  object
 1   Destination_airport  34775 non-null  object
 2   Distance             34775 non-null  int64 
dtypes: int64(1), object(2)
memory usage: 1.1+ MB


In [106]:
graph_data.head(5)

Origin_airport Destination_airport  Distance
1             EUG                 RDM       103
4             MFR                 RDM       156
9             SEA                 RDM       228
14            PDX                 RDM       116
38            LMT                 RDM       147

## Dataset 2: from which longitude and latitude information of airports can be extracted

This is done by combining origin and destination airport location information and then removing duplicate values.

In [99]:
org_data = data.drop(
    ['Destination_airport', 'Distance', 'Dest_airport_lat', 'Dest_airport_long'], axis=1)
org_data.rename(columns={'Origin_airport': 'airport',
                         'Org_airport_lat': 'lat', 'Org_airport_long': 'long'},
                inplace=True)
org_data.head(5)

airport        lat        long
1      EUG  44.124599 -123.211998
4      MFR  42.374199 -122.873001
9      SEA  47.449001 -122.308998
14     PDX  45.588699 -122.598000
38     LMT  42.156101 -121.733002

In [100]:
dest_data = data.drop(['Origin_airport', 'Distance',
                       'Org_airport_lat', 'Org_airport_long'], axis=1)
dest_data.rename(columns={'Destination_airport': 'airport', 'Dest_airport_lat': 'lat',
                          'Dest_airport_long': 'long'}, inplace=True)
dest_data.head(5)

airport        lat        long
1      RDM  44.254101 -121.150002
4      RDM  44.254101 -121.150002
9      RDM  44.254101 -121.150002
14     RDM  44.254101 -121.150002
38     RDM  44.254101 -121.150002

In [101]:
sum_airports = pd.concat([org_data, dest_data])

In [102]:
sum_airports.drop_duplicates(subset={'airport'}, inplace=True)

In [103]:
sum_airports.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 483 entries, 1 to 3138489
Data columns (total 3 columns):
 #   Column   Non-Null Count  Dtype  
---  ------   --------------  -----  
 0   airport  483 non-null    object 
 1   lat      483 non-null    float64
 2   long     483 non-null    float64
dtypes: float64(2), object(1)
memory usage: 15.1+ KB


In [104]:
sum_airports.head(5)

airport        lat        long
1      EUG  44.124599 -123.211998
4      MFR  42.374199 -122.873001
9      SEA  47.449001 -122.308998
14     PDX  45.588699 -122.598000
38     LMT  42.156101 -121.733002

## Store processed data as pickled objects

In [107]:
# dataset 1
graph_data.to_pickle('./airport-distance-data.pkl')

In [108]:
# dataset 2
sum_airports.to_pickle('./airport-geo-location-data.pkl')